riscv/triggers.cc

```diff
@@ -215,7 +215,7 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
   if (simple_match(xlen, value)) {
     /* This is OK because this function is only called if the trigger was not
      * inhibited by the previous trigger in the chain. */
-    hit = true;
+    set_hit(timing ? HIT_IMMEDIATELY_AFTER : HIT_BEFORE);
```

Comment:
Does this mean that for mcontrol6 triggers, timing is always immediately_after? That's a compliant implementation, but it would be better to implement the suggested timings:
![image](https://github.com/riscv-software-src/riscv-isa-sim/assets/16123885/901c4c24-fd09-404a-9af3-fc0de18b14b9)